dispatcher/MongoConnect.py

```diff
@@ -199,13 +203,20 @@ def AggregateStatus(self):
                     pass
 
                 try:
-                    status = STATUS(doc['status'])
+                    status = DAQ_STATUS(doc['status'])
                     dt = (now - int(str(doc['_id'])[:8], 16))
                     if dt > self.timeout:
                         self.log.debug('%s reported %i sec ago' % (doc['host'], int(dt)))
-                        status = STATUS.TIMEOUT
+                        status = DAQ_STATUS.TIMEOUT
+                        if self.host_config[doc['host']] == 'tpc':
+                            if (dt > self.timeout_take_action or
+                                    ((ts := self.HostAckdCommand(doc['host'])) is not None and
+                                     ts-now > self.timeout)):
+                                self.log.info(f'{doc["host"] is getting restarted')
+                                self.hypervisor.HandleTimeout(doc['host'])
```

Comment:
I've renamed this to `handle_timeout` in daqnt, sometimes I cannot help myself from having arbitrary preferences

Reply:
Dan initially named all the functions, I blame him for CamelCasing everything